src/archive/sevenz.rs

```diff
@@ -150,7 +159,51 @@ where
         }
 
         Ok(true)
-    })?;
+    };
+
+    let password = password.as_ref().map(|p| p.as_ref());
+
+    match password {
+        Some(password) => sevenz_rust::decompress_with_extract_fn_and_password(
+            reader,
+            output_path,
+            sevenz_rust::Password::from(password.to_str().unwrap()),
```

Comment:
inside of the `Some` branch of this `match`, we should try converting the provided bytes to UTF-8 or throw a small `FinalError` with the brief description that `sevenz_rust` expects utf-8 valid passwords.

you can do this with a `let-else` construct with a return inside